hazelcast/src/main/java/com/hazelcast/jet/impl/JetInstanceImpl.java

```diff
@@ -70,21 +70,27 @@ public Address getMasterId() {
     @Override
     public Map<Address, GetJobIdsResult> getJobsInt(String onlyName, Long onlyJobId) {
         Map<Address, CompletableFuture<GetJobIdsResult>> futures = new HashMap<>();
-        Address masterAddress = null;
+        Address masterAddress = nodeEngine.getMasterAddress();
```

Comment:
This value can be state as soon as it is read here. The code below will not be correct if master changes while it's executing. 

This approach could be better: we should send two `GetJobIdsOperation` to master: one as target invocation, fetching only light jobs, and another as a master invocation, checking for normal jobs. We will ignore errors for the one for master, no duplication of light jobs will occur if master is changed to another member, to which we also sent a request for light jobs and there will be no races between the operation and the mastership changes.

Reply:
Done